cast/src/lib.rs

```diff
@@ -443,6 +445,62 @@ where
         Ok(self.provider.get_gas_price().await?)
     }
 
+    pub async fn ls(&self) -> Result<String> {
+        // create a vector of addresses
+        let mut addrs: Vec<H160> = Vec::new();
+
+        // fetch the chain id
+        let chain_id = self.chain_id().await?;
+
+        // for each file in $HOME/.ethereum/keystore
+        if let Ok(home_dir) = std::env::var("HOME") {
```

Comment:
there's the `dirs-next` crate that provides a helper for finding the home dir
https://crates.io/crates/dirs-next

Reply:
See foundry_config crate for reference